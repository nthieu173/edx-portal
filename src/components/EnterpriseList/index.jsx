import React from 'react';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import Helmet from 'react-helmet';
import qs from 'query-string';
import isEmpty from 'lodash/isEmpty';
import omitBy from 'lodash/omitBy';

import H1 from '../H1';
import StatusAlert from '../StatusAlert';
import SearchBar from '../SearchBar';
import LoadingMessage from '../LoadingMessage';
import TableWithPagination from '../TableWithPagination';

import { formatTableOptions } from '../../utils';

class EnterpriseList extends React.Component {
  constructor(props) {
    super(props);

    const { enterprises, location } = props;

    const queryParams = this.formatQueryParams(qs.parse(location.search));

    this.state = {
      columns: [
        {
          label: 'Enterprise',
          key: 'link',
        },
      ],
      enterprises: enterprises && enterprises.results,
      pageCount: enterprises && enterprises.num_pages,
      searchQuery: queryParams.search || '',
      searchSubmitted: !!queryParams.search,
    };

    this.formatEnterpriseData = this.formatEnterpriseData.bind(this);
  }

  componentDidMount() {
    const options = qs.parse(this.props.location.search);
    this.clearPortalConfiguration();
    this.getEnterpriseList(formatTableOptions(options));
    this.props.getLocalUser();
  }

  componentDidUpdate(prevProps) {
    const { enterprises, searchQuery } = this.props;

    if (enterprises !== prevProps.enterprises) {
      this.setState({ // eslint-disable-line react/no-did-update-set-state
        enterprises: enterprises && enterprises.results,
        pageCount: enterprises && enterprises.num_pages,
      });
    }
    if (searchQuery !== prevProps.searchQuery) {
      const options = this.formatQueryParams({ search: searchQuery });
      this.props.history.push(`?${qs.stringify(options)}`);
      this.getEnterpriseList(options);
    }
  }

  getEnterpriseList(options) {
    this.props.getEnterpriseList(options);
  }

  clearPortalConfiguration() {
    this.props.clearPortalConfiguration();
  }

  formatQueryParams(options) {
    return omitBy({
      search: options.search,
    }, isEmpty);
  }

  formatEnterpriseData(enterprises) {
    if (!enterprises) {
      return null;
    } else if (!enterprises.length) {
      return [];
    }

    return enterprises.map(enterprise => ({
      link: <Link to={`/${enterprise.slug}/admin/learners`}>{enterprise.name}</Link>,
      name: enterprise.name,
      slug: enterprise.slug,
      uuid: enterprise.uuid,
    }));
  }

  handleSearch(query) {
    this.props.setSearchQuery(query);
    this.setState({
      searchQuery: query,
      searchSubmitted: true,
    });
  }

  handleClear() {
    this.props.setSearchQuery('');
    this.setState({
      searchQuery: '',
    });
  }

  shouldRenderRedirectToEnterpriseAdminPage() {
    const { loading, error } = this.props;
    const { enterprises, pageCount, searchSubmitted } = this.state;
    return (
      !loading && !error && enterprises && enterprises.length === 1 &&
        pageCount === 1 && !searchSubmitted
    );
  }

  renderEmptyEnterpriseListMessage() {
    const { searchQuery } = this.props;
    let emptyMessage = 'You do not have permission to view any enterprise customer data.';
    if (searchQuery) {
      emptyMessage = 'There are no enterprise customers that match your search.';
    }
    return (
      <StatusAlert
        alertType="warning"
        iconClassName={['fa', 'fa-exclamation-circle']}
        message={emptyMessage}
      />
    );
  }

  renderErrorMessage() {
    return (
      <StatusAlert
        alertType="danger"
        iconClassName={['fa', 'fa-times-circle']}
        title="Unable to load enterprise list"
        message={`Try refreshing your screen (${this.props.error.message})`}
      />
    );
  }

  renderRedirectToEnterpriseAdminPage() {
    const { enterprises } = this.state;
    return (
      <Redirect to={`/${enterprises[0].slug}/admin/learners`} />
    );
  }

  renderTableContent() {
    const {
      columns,
      enterprises,
      pageCount,
    } = this.state;

    return (
      <TableWithPagination
        columns={columns}
        data={enterprises}
        pageCount={pageCount}
        paginationLabel="enterprise list pagination"
        handleDataUpdate={options =>
          this.getEnterpriseList(options)
        }
        formatData={this.formatEnterpriseData}
      />
    );
  }

  renderLoadingMessage() {
    return <LoadingMessage className="enterprise-list" />;
  }

  render() {
    const { loading, error } = this.props;
    const { enterprises, searchQuery } = this.state;
    return (
      <div>
        <Helmet>
          <title>Enterprise List</title>
        </Helmet>
        <div className="container">
          <div className="row mt-4">
            <div className="col-sm-12 col-md">
              <H1>Enterprise List</H1>
            </div>
            <div className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0">
              <SearchBar
                onSearch={query => this.handleSearch(query)}
                onClear={() => this.handleClear()}
                value={searchQuery}
              />
            </div>
          </div>
          <div className="row mt-2">
            <div className="col">
              {error && this.renderErrorMessage()}
              {loading && !enterprises && this.renderLoadingMessage()}
              {!loading && !error && enterprises && enterprises.length === 0 &&
                this.renderEmptyEnterpriseListMessage()
              }
              {this.shouldRenderRedirectToEnterpriseAdminPage() &&
                this.renderRedirectToEnterpriseAdminPage()
              }
              {enterprises && enterprises.length > 0 && this.renderTableContent()}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

EnterpriseList.defaultProps = {
  enterprises: null,
  error: null,
  loading: false,
  location: {
    search: null,
  },
  searchQuery: null,
  history: {
    push: () => {},
  },
};

EnterpriseList.propTypes = {
  getEnterpriseList: PropTypes.func.isRequired,
  clearPortalConfiguration: PropTypes.func.isRequired,
  getLocalUser: PropTypes.func.isRequired,
  setSearchQuery: PropTypes.func.isRequired,
  enterprises: PropTypes.shape({
    count: PropTypes.number,
    num_pages: PropTypes.number,
    current_page: PropTypes.number,
    results: PropTypes.array,
    next: PropTypes.string,
    previous: PropTypes.string,
    start: PropTypes.number,
  }),
  loading: PropTypes.bool,
  error: PropTypes.instanceOf(Error),
  location: PropTypes.shape({
    search: PropTypes.string,
  }),
  searchQuery: PropTypes.string,
  history: PropTypes.shape({
    push: PropTypes.func,
  }),
};

export default EnterpriseList;
