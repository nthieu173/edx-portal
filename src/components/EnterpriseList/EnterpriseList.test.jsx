import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter, Redirect } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import EnterpriseList from './index';
import mockEnterpriseList from './EnterpriseList.mocks';

const EnterpriseListWrapper = props => (
  <MemoryRouter>
    <EnterpriseList
      getEnterpriseList={() => {}}
      clearPortalConfiguration={() => {}}
      getLocalUser={() => {}}
      setSearchQuery={() => {}}
      {...props}
    />
  </MemoryRouter>
);

describe('<EnterpriseList />', () => {
  let wrapper;

  describe('renders correctly', () => {
    it('calls getEnterpriseList, clearPortalConfiguration and getLocalUser props', () => {
      const mockGetEnterpriseList = jest.fn();
      const mockClearPortalConfiguration = jest.fn();
      const mockGetLocalUser = jest.fn();
      const tree = renderer
        .create((
          <EnterpriseListWrapper
            getEnterpriseList={mockGetEnterpriseList}
            clearPortalConfiguration={mockClearPortalConfiguration}
            getLocalUser={mockGetLocalUser}
          />
        ))
        .toJSON();
      expect(mockGetEnterpriseList).toHaveBeenCalled();
      expect(mockClearPortalConfiguration).toHaveBeenCalled();
      expect(mockGetLocalUser).toHaveBeenCalled();
      expect(tree).toMatchSnapshot();
    });

    it('with enrollment data', () => {
      const tree = renderer
        .create((
          <EnterpriseListWrapper
            enterprises={mockEnterpriseList}
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with empty enrollment data', () => {
      const tree = renderer
        .create((
          <EnterpriseListWrapper
            enterprises={{
              ...mockEnterpriseList,
              count: 0,
              results: [],
            }}
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with search query and empty enrollment data', () => {
      const tree = renderer
        .create((
          <EnterpriseListWrapper
            enterprises={{
              ...mockEnterpriseList,
              count: 0,
              results: [],
            }}
            searchQuery="enterprise name"
            searchSubmitted
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with error state', () => {
      const tree = renderer
        .create((
          <EnterpriseListWrapper error={Error('Network Error')} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with loading state', () => {
      const tree = renderer
        .create((
          <EnterpriseListWrapper loading />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('redirects when there is only one enterprise', () => {
      const oneEnterpriseListData = {
        count: 1,
        current_page: 1,
        num_pages: 1,
        next: null,
        previous: null,
        results: [{
          uuid: 'ee5e6b3a-069a-4947-bb8d-d2dbc323396c',
          name: 'Enterprise 99',
          slug: 'enterprise-99',
        }],
        start: 0,
      };
      wrapper = mount((
        <MemoryRouter initialEntries={['/test']}>
          <EnterpriseList
            enterprises={oneEnterpriseListData}
            getEnterpriseList={() => {}}
            clearPortalConfiguration={() => {}}
            getLocalUser={() => {}}
            setSearchQuery={() => {}}
          />
        </MemoryRouter>
      ));
      const expectedRedirect = <Redirect to="/enterprise-99/admin/learners" />;
      expect(wrapper.containsMatchingElement(expectedRedirect)).toEqual(true);
    });
  });

  describe('formatEnterpriseData', () => {
    const expectedData = mockEnterpriseList.results;

    beforeEach(() => {
      wrapper = shallow((
        <EnterpriseListWrapper
          enterprises={mockEnterpriseList}
        />
      )).find(EnterpriseList);
    });

    it('overrides state enterprises when props enterprises changes with new enterprises', () => {
      const currentState = wrapper.dive().state();

      expect(currentState.enterprises[0]).toEqual(expectedData[0]);
      expect(currentState.enterprises[1]).toEqual(expectedData[1]);
      expect(currentState.pageCount).toEqual(mockEnterpriseList.num_pages);

      wrapper.dive().setProps({
        enterprises: {
          count: 0,
          num_pages: 0,
          results: [],
        },
      }, () => {
        expect(wrapper.dive().state('enterprises')).toEqual([]);
        expect(wrapper.dive().state('pageCount')).toEqual(null);
      });
    });

    it('does not override state enterprises when props enterprises changes with existing enterprises', () => {
      const currentEnterprises = wrapper.dive().state('enterprises');
      expect(currentEnterprises[0]).toEqual(expectedData[0]);
      expect(currentEnterprises[1]).toEqual(expectedData[1]);

      wrapper.dive().setProps({
        enterprises: mockEnterpriseList,
      });

      const updatedEnterprises = wrapper.dive().state('enterprises');
      expect(updatedEnterprises[0]).toEqual(expectedData[0]);
      expect(updatedEnterprises[1]).toEqual(expectedData[1]);
    });
  });

  it('pageCount state should be null when enterprises prop is null', () => {
    wrapper = shallow((
      <EnterpriseListWrapper
        enterprises={null}
      />
    )).find('EnterpriseList');
    expect(wrapper.dive().state('pageCount')).toEqual(null);
  });

  describe('enterprise list search', () => {
    const submitSearch = (searchQuery) => {
      wrapper.find('SearchBar').find('input[type=\'search\']').simulate('change', { target: { value: searchQuery } });
      expect(wrapper.find('SearchBar').find('input[type=\'search\']').prop('value')).toEqual(searchQuery);
      wrapper.find('SearchBar').find('.input-group-append').find('button').last()
        .simulate('click');
      expect(wrapper.find('EnterpriseList').instance().state.searchQuery).toEqual(searchQuery);
      expect(wrapper.find('EnterpriseList').instance().state.searchSubmitted).toBeTruthy();
    };

    it('searchQuery state changes onSearch', () => {
      wrapper = mount((
        <EnterpriseListWrapper
          enterprises={mockEnterpriseList}
        />
      ));

      submitSearch('Enterprise 1');
    });

    it('searchQuery state changes onClear', () => {
      wrapper = mount((
        <EnterpriseListWrapper
          enterprises={mockEnterpriseList}
        />
      ));

      submitSearch('Enterprise 1');

      wrapper.find('SearchBar').find('.input-group-append').find('button').first()
        .simulate('click');
      expect(wrapper.find('SearchBar').find('input[type=\'search\']').prop('value')).toEqual('');
      expect(wrapper.find('EnterpriseList').instance().state.searchQuery).toEqual('');
    });

    it('dispatches enterprise list request action when searchQuery prop changes', () => {
      const searchQuery = 'enterprise name';
      const mockGetEnterpriseList = jest.fn();
      wrapper = shallow((
        <EnterpriseListWrapper
          enterprises={mockEnterpriseList}
          getEnterpriseList={mockGetEnterpriseList}
        />
      )).find('EnterpriseList');

      wrapper.dive().setProps({
        searchQuery,
      }, () => {
        expect(wrapper.dive().state('searchQuery')).toEqual(searchQuery);
        expect(mockGetEnterpriseList).toHaveBeenCalledTimes(1);
      });
    });
  });
});
