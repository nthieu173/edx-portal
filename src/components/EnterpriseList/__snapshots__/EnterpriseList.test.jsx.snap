// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnterpriseList /> renders correctly calls getEnterpriseList, clearPortalConfiguration and getLocalUser props 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col-sm-12 col-md"
      >
        <h1
          className={null}
        >
          Enterprise List
        </h1>
      </div>
      <div
        className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0"
      >
        <div
          className="border search-field"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="form-group form-inline"
          >
            <label
              className=""
              htmlFor="asInput2"
              id="label-asInput2"
            >
              Search:
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="error-asInput2"
                aria-invalid={false}
                autoComplete="off"
                className="form-control is-invalid-nodanger input no-clear-btn"
                disabled={false}
                id="asInput2"
                name="search"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                themes={Array []}
                type="search"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn search-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span>
                    <span
                      aria-hidden={true}
                      className="fa fa-search"
                      id="Icon1"
                    />
                    <span
                      className="sr-only"
                    >
                      Submit search
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-live="polite"
              className="invalid-feedback invalid-feedback-nodanger"
              id="error-asInput2"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-2"
    >
      <div
        className="col"
      />
    </div>
  </div>
</div>
`;

exports[`<EnterpriseList /> renders correctly with empty enrollment data 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col-sm-12 col-md"
      >
        <h1
          className={null}
        >
          Enterprise List
        </h1>
      </div>
      <div
        className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0"
      >
        <div
          className="border search-field"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="form-group form-inline"
          >
            <label
              className=""
              htmlFor="asInput6"
              id="label-asInput6"
            >
              Search:
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="error-asInput6"
                aria-invalid={false}
                autoComplete="off"
                className="form-control is-invalid-nodanger input no-clear-btn"
                disabled={false}
                id="asInput6"
                name="search"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                themes={Array []}
                type="search"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn search-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span>
                    <span
                      aria-hidden={true}
                      className="fa fa-search"
                      id="Icon1"
                    />
                    <span
                      className="sr-only"
                    >
                      Submit search
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-live="polite"
              className="invalid-feedback invalid-feedback-nodanger"
              id="error-asInput6"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-2"
    >
      <div
        className="col"
      >
        <div
          className="alert fade alert-warning show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="d-flex"
            >
              <div
                className="icon mr-2"
              >
                <span>
                  <span
                    aria-hidden={true}
                    className="fa fa-exclamation-circle"
                    id="Icon1"
                  />
                </span>
              </div>
              <div
                className="message"
              >
                <span>
                  You do not have permission to view any enterprise customer data.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<EnterpriseList /> renders correctly with enrollment data 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col-sm-12 col-md"
      >
        <h1
          className={null}
        >
          Enterprise List
        </h1>
      </div>
      <div
        className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0"
      >
        <div
          className="border search-field"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="form-group form-inline"
          >
            <label
              className=""
              htmlFor="asInput3"
              id="label-asInput3"
            >
              Search:
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="error-asInput3"
                aria-invalid={false}
                autoComplete="off"
                className="form-control is-invalid-nodanger input no-clear-btn"
                disabled={false}
                id="asInput3"
                name="search"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                themes={Array []}
                type="search"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn search-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span>
                    <span
                      aria-hidden={true}
                      className="fa fa-search"
                      id="Icon1"
                    />
                    <span
                      className="sr-only"
                    >
                      Submit search
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-live="polite"
              className="invalid-feedback invalid-feedback-nodanger"
              id="error-asInput3"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-2"
    >
      <div
        className="col"
      >
        <div
          className={null}
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-sm table-striped"
                >
                  <thead
                    className=""
                  >
                    <tr
                      className=""
                    >
                      <th
                        className=""
                        scope="col"
                      >
                        Enterprise
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className=""
                  >
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-1/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 1
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-2/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 2
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-3/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 3
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-4/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 4
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-5/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 5
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-6/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 6
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-7/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 7
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-8/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 8
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-9/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 9
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-10/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 10
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-11/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 11
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-12/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 12
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-13/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 13
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-14/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 14
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-15/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 15
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-16/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 16
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-17/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 17
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-18/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 18
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-19/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 19
                        </a>
                      </td>
                    </tr>
                    <tr
                      className=""
                    >
                      <td
                        className=""
                      >
                        <a
                          href="/enterprise-20/admin/learners"
                          onClick={[Function]}
                        >
                          Enterprise 20
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            className="row mt-2"
          >
            <div
              className="col d-flex justify-content-center"
            >
              <nav
                aria-label="enterprise list pagination"
                className=""
              >
                <div
                  aria-atomic={true}
                  aria-live="polite"
                  aria-relevant="text"
                  className="sr-only"
                >
                  Page 1, Current Page, of 2
                </div>
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item disabled"
                  >
                    <button
                      aria-label="Previous"
                      className="btn previous page-link"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      tabIndex="-1"
                      type="button"
                    >
                      <div>
                        <span>
                          <span
                            aria-hidden={true}
                            className="fa fa-chevron-left mr-2"
                            id="pagination-4"
                          />
                        </span>
                        Previous
                      </div>
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      aria-label="Next, Page 2"
                      className="btn next page-link"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      tabIndex={undefined}
                      type="button"
                    >
                      <div>
                        Next
                        <span>
                          <span
                            aria-hidden={true}
                            className="fa fa-chevron-right ml-2"
                            id="pagination-5"
                          />
                        </span>
                      </div>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<EnterpriseList /> renders correctly with error state 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col-sm-12 col-md"
      >
        <h1
          className={null}
        >
          Enterprise List
        </h1>
      </div>
      <div
        className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0"
      >
        <div
          className="border search-field"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="form-group form-inline"
          >
            <label
              className=""
              htmlFor="asInput8"
              id="label-asInput8"
            >
              Search:
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="error-asInput8"
                aria-invalid={false}
                autoComplete="off"
                className="form-control is-invalid-nodanger input no-clear-btn"
                disabled={false}
                id="asInput8"
                name="search"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                themes={Array []}
                type="search"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn search-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span>
                    <span
                      aria-hidden={true}
                      className="fa fa-search"
                      id="Icon1"
                    />
                    <span
                      className="sr-only"
                    >
                      Submit search
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-live="polite"
              className="invalid-feedback invalid-feedback-nodanger"
              id="error-asInput8"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-2"
    >
      <div
        className="col"
      >
        <div
          className="alert fade alert-danger show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="d-flex"
            >
              <div
                className="icon mr-2"
              >
                <span>
                  <span
                    aria-hidden={true}
                    className="fa fa-times-circle"
                    id="Icon1"
                  />
                </span>
              </div>
              <div
                className="message"
              >
                <span
                  className="title"
                >
                  Unable to load enterprise list
                </span>
                <span>
                  Try refreshing your screen (Network Error)
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<EnterpriseList /> renders correctly with loading state 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col-sm-12 col-md"
      >
        <h1
          className={null}
        >
          Enterprise List
        </h1>
      </div>
      <div
        className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0"
      >
        <div
          className="border search-field"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="form-group form-inline"
          >
            <label
              className=""
              htmlFor="asInput9"
              id="label-asInput9"
            >
              Search:
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="error-asInput9"
                aria-invalid={false}
                autoComplete="off"
                className="form-control is-invalid-nodanger input no-clear-btn"
                disabled={false}
                id="asInput9"
                name="search"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                themes={Array []}
                type="search"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn search-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span>
                    <span
                      aria-hidden={true}
                      className="fa fa-search"
                      id="Icon1"
                    />
                    <span
                      className="sr-only"
                    >
                      Submit search
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-live="polite"
              className="invalid-feedback invalid-feedback-nodanger"
              id="error-asInput9"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-2"
    >
      <div
        className="col"
      >
        <div
          className="enterprise-list loading d-flex align-items-center justify-content-center"
        >
          <span>
            Loading...
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<EnterpriseList /> renders correctly with search query and empty enrollment data 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col-sm-12 col-md"
      >
        <h1
          className={null}
        >
          Enterprise List
        </h1>
      </div>
      <div
        className="col-sm-12 col-md-6 col-lg-4 mb-3 mb-md-0"
      >
        <div
          className="border search-field"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="form-group form-inline"
          >
            <label
              className=""
              htmlFor="asInput7"
              id="label-asInput7"
            >
              Search:
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              />
              <input
                aria-describedby="error-asInput7"
                aria-invalid={false}
                autoComplete="off"
                className="form-control is-invalid-nodanger input no-clear-btn"
                disabled={false}
                id="asInput7"
                name="search"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                themes={Array []}
                type="search"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn search-btn"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span>
                    <span
                      aria-hidden={true}
                      className="fa fa-search"
                      id="Icon1"
                    />
                    <span
                      className="sr-only"
                    >
                      Submit search
                    </span>
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-live="polite"
              className="invalid-feedback invalid-feedback-nodanger"
              id="error-asInput7"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-2"
    >
      <div
        className="col"
      >
        <div
          className="alert fade alert-warning show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="d-flex"
            >
              <div
                className="icon mr-2"
              >
                <span>
                  <span
                    aria-hidden={true}
                    className="fa fa-exclamation-circle"
                    id="Icon1"
                  />
                </span>
              </div>
              <div
                className="message"
              >
                <span>
                  There are no enterprise customers that match your search.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
